src/btree/bt_read.c

```diff
@@ -332,30 +407,43 @@ __las_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
              * replacing it with.
              */
             while (upd_type == WT_UPDATE_MODIFY) {
-                if (mod_counter >= WT_MODIFY_ARRAY_SIZE) {
-                    if (mod_counter == WT_MODIFY_ARRAY_SIZE)
-                        listp = NULL;
-                    WT_ERR(__wt_realloc_def(session, &allocated_bytes, mod_counter + 1, &listp));
-                    if (mod_counter == WT_MODIFY_ARRAY_SIZE)
-                        memcpy(listp, list, sizeof(list));
-                }
-                WT_ERR(__wt_update_alloc(
-                  session, &las_value, &listp[mod_counter++], &notused, upd_type));
-                WT_ERR(cursor->prev(cursor));
-                WT_ERR(cursor->get_value(
-                  cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+                WT_ERR(__wt_update_alloc(session, &las_value, &mod_upd, &notused, upd_type));
+                WT_ERR(__wt_modify_vector_push(&modifies, mod_upd));
+                mod_upd = NULL;
+                WT_ERR(las_cursor->prev(las_cursor));
```

Comment:
Isn't the beginning of the table (`WT_NOTFOUND`) also a boundary condition here?

Reply:
1dc2b0c

Thanks for picking up on that. I've added some logic to treat this case the same.